### ISSUE TITLE
Bug: using an alias in module names

### ISSUE THREAD

**Issue by Joon-Klaps** [askimed/nf-test#139]
Nf-test (nf-test 0.8.1) doesn't support the aliassing of modules in the `setup` step correctly.

To create a test for bbmap/bbsplit I run the same module twice (as was done for the original pytest). 

The problem here is that I need to create an alias for the module. 
```nextflow
include {BBMAP_BBSPLIT as BBMAP_BBSPLIT_INDEX}  ...
``` 
It is correctly importing this but not correctly calling it see the mock.nf for the complete script 

```nextflow
...
BBMAP_BBSPLIT as BBMAP_BBSPLIT_INDEX(*input)
...
```
```Error
Nextflow stdout:
  
  ERROR ~ No such variable: as
  
   -- Check script 'meta/mock.nf' at line: 47 or see '/workspace/modules/.nf-test/tests/883826a09088b68f2eb51c0096a941d8/meta/nextflow.log' file for more details
  Nextflow stderr:
 ```

[Archive.zip](https://github.com/askimed/nf-test/files/12999568/Archive.zip)

or see the PR https://github.com/nf-core/modules/pull/4131

**Comment by apeltzer** [comment]
Yes would be really helpful to reduce the amount of code you have to duplicate. Example would be umicollapse in nf-core, where I just had to copy two identical setup() blocks for stub testing and PE testing into the respective tests as otherwise their import names were conflicting with the one from the SE test.

https://github.com/nf-core/modules/pull/4799

**Comment by jasmezz** [comment]
Agree that aliasing should be implented asap :hand:
E.g. for [this module](https://github.com/jasmezz/modules/commit/e56ac5d5863b5c88b09c2648beb1319f3b9b9923) we need to untar 3 archives from its test data which is now impossible (and a module update is critical for the [funcscan](https://github.com/nf-core/funcscan) pipeline :pleading_face:)

**Comment by lukfor** [comment]
Thanks @apeltzer for the feedback! What do you think about using an optional `alias` property in the `run` method? This would be more robust than parsing `run("GENERATE_DATA as LUKAS")`.  Would this help in eliminating duplicate code? 

```groovy
nextflow_process {

    name "Test process data"

    script "./process_data.nf"
    process "PROCESS_DATA"

    test("Should use process GENERATE_DATA with alias LUKAS to generate input data") {

        setup {
            run("GENERATE_DATA", alias: "LUKAS") {
                script "./generate_data.nf"
                process {
                    """
                    input[0] = "nf-core"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = "Hey "
                input[1] = LUKAS.out.results
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.results).match()
        }
    }

}
```

I am not sure if the same mechanism makes sense for the process/workflow name it self (see #96). Do you have a usecase in mind where it makes sense to give the test subject a different alias?
